### ISSUE TITLE
xpath-suppression isn't compatible with any ...NameCheck

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#5477]
`/var/tmp $ cat TestClass.java`
```
import java.util.Arrays;
import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier;

public class TestClass {
    private boolean matchAccessModifiers1(final AccessModifier accessModifier) {
        return Arrays.stream(accessModifiers).anyMatch((el) -> el == accessModifier);
    }

    private void methodParameter(AccessModifier  el) {
    }
}
```

`/var/tmp $ cat config.xml`
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
      <module name="SuppressionXpathFilter">
        <property name="file" value="suppressions.xml"/>
      </module>
      <module name="ParameterName">
        <property name="format" value="^(id)|([a-z][a-z0-9][a-zA-Z0-9]+)$"/>
      </module>
    </module>
</module>
```

`/var/tmp $ cat suppressions.xml`
```
<?xml version="1.0"?>
<!DOCTYPE suppressions PUBLIC
    "-//Puppy Crawl//DTD Suppressions Xpath Experimental 1.2//EN"
    "http://checkstyle.sourceforge.net/dtds/suppressions_1_2_xpath_experimental.dtd">
<suppressions>
    <suppress-xpath checks="ParameterName"
                    query="//LAMBDA/*"/>
</suppressions>
```

`/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.7-all.jar -c TestConfig.xml TestClass.java`

```
Starting audit...
[ERROR] /tmp/TestClass.java:6:56: Name 'el' must match pattern '^(id)|([a-z][a-z0-9][a-zA-Z0-9]+)$'. [ParameterName]
[ERROR] /tmp/TestClass.java:9:49: Name 'el' must match pattern '^(id)|([a-z][a-z0-9][a-zA-Z0-9]+)$'. [ParameterName]
Audit done.
Checkstyle ends with 2 errors.

```
---------------

Only one violation is expected. The bug is here (XpathFilter::isXpathQueryMatching)
```
                isMatching = abstractNode.getTokenType() == event.getTokenType()
                        && abstractNode.getLineNumber() == event.getLine()
                        && abstractNode.getColumnNumber() == event.getColumnCharIndex();

```
abstractNode.getColumnNumber() is zero-based, but event.getColumnCharIndex() is one-based.
Seems to be an easy task.

**Comment by pbludov** [comment]
The bug actually here `AbstractCheck#log(int, int, java.lang.String, java.lang.Object...)`
```
    @Override
    public final void log(int lineNo, int colNo, String key,
            Object... args) {
        final int col = 1 + CommonUtils.lengthExpandedTabs(
            getLines()[lineNo - 1], colNo, tabWidth);
        context.get().messages.add(
            new LocalizedMessage(
                lineNo,
                col,
                getMessageBundle(),
                key,
                args,
                getSeverityLevel(),
                getId(),
                getClass(),
                getCustomMessages().get(key)));
    }
```
it should be
```
    @Override
    public final void log(int lineNo, int colNo, String key,
            Object... args) {
        final int col = 1 + CommonUtils.lengthExpandedTabs(
            getLines()[lineNo - 1], colNo, tabWidth);
        context.get().messages.add(
            new LocalizedMessage(
                lineNo,
                col,
                colNo,  // Missing one.
                0,
                getMessageBundle(),
                key,
                args,
                getSeverityLevel(),
                getId(),
                getClass(),
                getCustomMessages().get(key)));
    }
```

**Comment by varunu28** [comment]
@pbludov Let me know if I can go ahead with the changes?

**Comment by MEZk** [comment]
@varunu28 

The official documentation clearly says that [SuppressionXpathFilter](http://checkstyle.sourceforge.net/config_filters.html#SuppressionXpathFilter) currenctly does not support that support ParameterNameCheck. Please, follow the documentation.

> for other Checks will be available after resolving issue #4830.

**Comment by rnveach** [comment]
I am closing this since there is no support.
When support is added and regression example is added, it will show the check works with xpath.
